src/big_widgets/GitQlient.cpp

```diff
@@ -110,67 +102,82 @@ void GitQlient::setRepositories(const QStringList &repositories)
 #else
-   logLevel = static_cast<LogLevel>(settings.globalValue("logsLevel", static_cast<int>(LogLevel::Warning)).toInt());
+   LogLevel logLevel = static_cast<LogLevel>(settings.globalValue("logsLevel", static_cast<int>(LogLevel::Warning)).toInt());
 #endif
+   bool areLogsDisabled = settings.globalValue("logsDisabled", true).toBool();
 
-   if (arguments.contains("-noLog") || settings.globalValue("logsDisabled", true).toBool())
-      QLoggerManager::getInstance()->pause();
-   else
-      QLoggerManager::getInstance()->overwriteLogLevel(logLevel);
+   QCommandLineParser parser;
+   parser.setApplicationDescription(tr("Graphical Git client"));
+   parser.addPositionalArgument("repos", tr("Git repositories to open"), tr("[repos...]"));
 
-   QLog_Info("UI", QString("Getting arguments {%1}").arg(arguments.join(", ")));
+   const QCommandLineOption helpOption = parser.addHelpOption();
+   // We don't use parser.addVersionOption() because then it is handled by Qt and we want to show Git SHA also
+   const QCommandLineOption versionOption(QStringList() << "v" << "version", tr("Displays version information."));
+   parser.addOption(versionOption);
 
-   QStringList repos;
-   const auto argSize = arguments.count();
+   const QCommandLineOption noLogOption("no-log", tr("Disables logs."));
+   parser.addOption(noLogOption);
+
+   const QCommandLineOption logLevelOption("log-level", tr("Sets log level."), tr("level"));
+   parser.addOption(logLevelOption);
+
+   parser.process(arguments);
 
-   for (auto i = 0; i < argSize;)
+   *repos = parser.positionalArguments();
+   if (parser.isSet(noLogOption))
+      areLogsDisabled = true;
+
+   if (!areLogsDisabled)
    {
-      if (arguments.at(i) == "-repos")
-      {
-         while (++i < argSize && !arguments.at(i).startsWith("-"))
-            repos.append(arguments.at(i));
-      }
-      else
+      if (parser.isSet(logLevelOption))
       {
-         if (arguments.at(i) == "-logLevel")
+         LogLevel level = static_cast<LogLevel>(parser.value(logLevelOption).toInt());
+         if (level >= QLogger::LogLevel::Trace && level <= QLogger::LogLevel::Fatal)
          {
-            logLevel = static_cast<LogLevel>(arguments.at(++i).toInt());
-
-            if (logLevel >= QLogger::LogLevel::Trace && logLevel <= QLogger::LogLevel::Fatal)
-            {
-               const auto logger = QLoggerManager::getInstance();
-               logger->overwriteLogLevel(logLevel);
-
-               settings.setGlobalValue("logsLevel", static_cast<int>(logLevel));
-            }
+            logLevel = level;
+            settings.setGlobalValue("logsLevel", static_cast<int>(level));
          }
-
-         ++i;
       }
+
+      QLoggerManager::getInstance()->overwriteLogLevel(logLevel);
+      QLog_Info("UI", QString("Getting arguments {%1}").arg(arguments.join(", ")));
+   }
+   else
+      QLoggerManager::getInstance()->pause();
+
+   if (parser.isSet(versionOption))
+   {
+      QTextStream out(stdout);
```

Comment:
Is it possible to use QString instead and throw the .data to std::cout instead?

If possible, I'd prefer to avoid the use of QTextStream since it's quite more expensive that QString + args.

Reply:
I realize now that it's stdout. Nevermind!

Reply:
Yes, I used QTextStream to have local aware stdout output.